src/app/iteration/iteration.service.ts

```diff
@@ -80,10 +81,35 @@ export class IterationService {
   }
 
   /**
-   * Get locally saved iterations' reference
+   * Update an existing iteration
+   * @param iteration - Updated iteration
+   * @return updated iteration's reference from the list
    */
-  getIterationList() {
-    return this.iterations;
+  updateIteration(iteration: IterationModel): Promise<IterationModel> {
+    return this.http
+      .patch(iteration.links.self, { headers: this.headers })
+      .toPromise()
+      .then (response => {
+        if (/^[5, 4][0-9]/.test(response.status.toString())) {
```

Comment:
I meant it does not make sense here to involve regex which is more expensive operation that just simple number comparison. Also the condition is more readable from my point of view - you immediately see what is the logic about.